src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,68 @@
 package core.basesyntax;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
 public class WorkWithFile {
+    private static final String DELIMITER = ",";
+    private static final String LINE_SEPARATOR = System.lineSeparator();
+    private static final int KEY_POSITION = 0;
+    private static final int VALUE_POSITION = 1;
+    private static final String SUPPLY_KEY = "supply";
+    private static final String BUY_KEY = "buy";
+    private static final String RESULT_STRING = "result";
 
     public void getStatistic(String fromFileName, String toFileName) {
+        Map<String, Integer> csvContent;
+        String report;
+        try {
+            csvContent = readCsvFile(Path.of(fromFileName));
+            report = generateReport(csvContent);
+        } catch (IOException | NumberFormatException e) {
+            throw new RuntimeException("Unable to read file: " + e);
+        }
+        try {
+            writeToCsv(Path.of(toFileName), report);
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to write file: " + e);
+        }
```

Comment:
Let's move exception handling as close as possible to the place where it can be thrown

Reply:
try-catch bloks moved to read/write methods.
+fixed in new commit